src/libraries/System.Drawing.Common/src/System/Drawing/Icon.Windows.cs

```diff
@@ -664,32 +664,23 @@ public void Save(Stream outputStream)
             }
         }
 
-        private void CopyBitmapData(BitmapData sourceData, BitmapData targetData)
+        private unsafe void CopyBitmapData(BitmapData sourceData, BitmapData targetData)
         {
-            int offsetSrc = 0;
-            int offsetDest = 0;
+            byte* srcPtr = (byte*)sourceData.Scan0;
+            byte* destPtr = (byte*)targetData.Scan0;
 
             Debug.Assert(sourceData.Height == targetData.Height, "Unexpected height. How did this happen?");
+            int height = Math.Min(sourceData.Height, targetData.Height);
+            long bytesToCopyEachIter = Math.Abs(targetData.Stride);
 
-            for (int i = 0; i < Math.Min(sourceData.Height, targetData.Height); i++)
+            for (int i = 0; i < height; i++)
             {
-                IntPtr srcPtr, destPtr;
-                if (IntPtr.Size == 4)
-                {
-                    srcPtr = new IntPtr(sourceData.Scan0.ToInt32() + offsetSrc);
-                    destPtr = new IntPtr(targetData.Scan0.ToInt32() + offsetDest);
-                }
-                else
-                {
-                    srcPtr = new IntPtr(sourceData.Scan0.ToInt64() + offsetSrc);
-                    destPtr = new IntPtr(targetData.Scan0.ToInt64() + offsetDest);
-                }
-
-                UnsafeNativeMethods.CopyMemory(new HandleRef(this, destPtr), new HandleRef(this, srcPtr), Math.Abs(targetData.Stride));
-
-                offsetSrc += sourceData.Stride;
-                offsetDest += targetData.Stride;
+                Buffer.MemoryCopy(srcPtr, destPtr, bytesToCopyEachIter, bytesToCopyEachIter);
```

Comment:
Why `Buffer.MemoryCopy` rather than `Span.CopyTo`?
Also, why copy by row rather than just the entire bitmap at once (which I would imagine has higher throughput)?

Reply:
`Buffer.MemoryCopy` because I'm already dealing with raw pointers, so may as well use the pointer-based APIs. Converting them to span would introduce artificial complexity for no benefit. It doesn't make the code "safe", for instance.

Copy by row because the source and the destination might have different stride lengths.

Reply:
In theory, if the destination stride length is shorter than the source stride length, the copy will still "succeed". Though subsequent row copies might overwrite previous row copies, and I don't know what happens to the data beyond the last row of the dest. Hence some of my nervousness around trying to infer what an appropriate "actual source length" parameter would be, so I'm continuing to say "eh, just assume the source is large enough" - which matches what the original code did. I'm not trying to fix any bugs that might exist in this code, just bring the code into compliance.

Reply:
> Copy by row because the source and the destination might have different stride lengths.

Ah, I didn't realize that this supported copying a subset of the original bitmap; that makes sense.